polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -433,8 +404,7 @@ CodeGenTypes::getFunctionType(const clang::CodeGen::CGFunctionInfo &FI,
       LLVM_DEBUG(llvm::dbgs() << "ArgInfo: ABIArgInfo::IndirectAliased\n");
       assert(NumIRArgs == 1);
       mlir::Type MLIRArgTy = getMLIRArgType(DeclArgTy);
-      ArgTypes[FirstIRArg] = mlir::MemRefType::get(
-          -1, MLIRArgTy, {}, ArgInfo.getIndirectAddrSpace());
+      ArgTypes[FirstIRArg] = getPointerOrMemRefType(MLIRArgTy);
```

Comment:
Pass address space `ArgInfo.getIndirectAddrSpace()`

Reply:
Note: in principle we should always create a memref here because we want to avoid lowering the abstraction level at this point in the compilation flow. However, `cgeist` treats type inconsistently, it expects `memref` for SYCL types and pointers for every other type... That is a terrible design IMHO, but for now we need to live with it. We should add a comment to the `getPointerOrMemRefType` function to indicate it is a temporary hack (and explain the reason).